Assets/MixedRealityToolkit/Extensions/GameObjectExtensions.cs

```diff
@@ -158,5 +158,47 @@ public static void DestroyGameObject(GameObject gameObject, float t = 0.0f)
         {
             UnityObjectExtensions.DestroyObject(gameObject, t);
         }
+
+        /// <summary>
+        /// Checks if any MonoBehaviour on the given GameObject is using the RequireComponentAttribute requiring type T
+        /// </summary>
+        /// <typeparam name="T">The potentially required component</typeparam>
+        /// <param name="gameObject">the GameObject requiring the component</param>
+        /// <param name="requiringTypes">A list of types that do require the component in question</param>
+        /// <returns></returns>
+        public static bool IsComponentRequired<T>(this GameObject gameObject, List<Type> requiringTypes) where T : Component
```

Comment:
Nit: It might be interesting to pass the requiringTypes as an out explicitly to be clear that it's a mechanism for returning state.

Usually these other things tend to be in parameters. Or at least, the name of the function doesn't suggest that it's mutating the requiringTypes. (i.e. if it were named GetDependentComponents it might be more obvious)